test/integration/cli-domain-package-server-script/cli-domain-package-server-script.js

```diff
@@ -0,0 +1,378 @@
+    describe('when component implements not-valid javascript', function(){
+      var serverContent = '\nmodule.exports.data=function(context,cb){\nreturn cb(null,data; };';
+
+      beforeEach(function(done){
+        fs.writeFileSync(path.resolve(componentPath, serverName), serverContent);
+        done();
+      });
+
+      it('should throw an error with error details', function(done){
+        try {
+          packageServerScript(
+            {
+              componentPath: componentPath,
+              ocOptions: {
+                files: {
+                  data: serverName
+                }
+              },
+              publishPath: publishPath,
+              webpack: webpackOptions
+            },
+            function(err, res){
+              try {
+                expect(err.toString()).to.contain.contain('Unexpected token, expected , (3:19)');
+                return done();
+              } catch(e) {
+                return done(e);
+              }
+              return done('error');
+            }
+          );
+        } catch (e) {
+          expect(e).to.contain.contain('Unexpected token, expected , (3:19)');
+          return done();
+        }
+      });
+    });
+
+    describe('when component does not require any json', function(){
+      var serverContent = '\nmodule.exports.data=function(context,cb){\nreturn cb(null, {name:\'John\'});\n};';
+
+      beforeEach(function(done){
+        fs.writeFileSync(path.resolve(componentPath, serverName), serverContent);
+        done();
+      });
+
+      it('should save compiled data provider and return a hash for the script', function(done){
+        packageServerScript(
+          {
+            componentPath: componentPath,
+            ocOptions: {
+              files: {
+                data: serverName
+              }
+            },
+            publishPath: publishPath,
+            webpack: webpackOptions
+          },
+          function(err, res){
+            if (err) {
+              return done(err);
+            }
+            try {
+              expect(res.type).to.equal('node.js');
+              expect(res.src).to.equal('server.js');
+
+              var compiledContent = fs.readFileSync(path.resolve(publishPath, res.src), {encoding: 'utf8'});
+              expect(res.hashKey).to.equal(hashBuilder.fromString(compiledContent));
+              done();
+            } catch(e) {
+              done(e);
+            }
+          }
+        );
+      });
+    });
+
+    describe('when component require a json file', function(){
+      var user = {first: 'John',last:'Doe'};
+      var jsonContent = JSON.stringify(user);
+      var serverContent = 'var user = require(\'./user\');\nmodule.exports.data=function(){return user.first;};';
+
+      beforeEach(function(done){
+        fs.writeFileSync(path.resolve(componentPath, 'user.json'), jsonContent);
+        fs.writeFileSync(path.resolve(componentPath, serverName), serverContent);
+        done();
+      });
```

Comment:
we should add an afterEach for ensuring that other requires to same (changed?) file don't get the cached version
```js
require.cache[path.resolve(publishPath, res.src)] = null;
```